float32 height
float32 z_velocity
